lib/data_access/data-store.service.ts

```diff
@@ -26,7 +26,7 @@ export class DataStoreService{
 		const fullHttpConfig:AjaxRequest = Object.assign({}, this.config.http, httpConfig);
 		const endpointUrl = this.getEndpointUrl(endpoint, baseUrl);
 
-		return this.setActiveRequest(Http.request(method, endpointUrl, data, fullHttpConfig), method, endpointUrl, data);
+		return this.setActiveRequest(Http.request(method, endpointUrl, data, fullHttpConfig, this.config.ajaxService), method, endpointUrl, data);
```

Comment:
Following the previous comment, don't pass the ajax service on each call, just set it on init.